### ISSUE TITLE
Evaluation of time series clustering

### ISSUE THREAD

**Issue by jonpappalord** [rtavenar/tslearn#14]
It would be nice to allow the user to evaluate the quality of a clustering by providing the equivalent of the silhouette score (or related metric) for time series clustering.

**Comment by rtavenar** [comment]
Hi,

This is a very good idea! I'll do that as soon as I can. If anyone is interested in giving a hand on this, the idea would be to adapt [the code from sklearn](https://github.com/scikit-learn/scikit-learn/blob/b661a9c8/sklearn/metrics/cluster/unsupervised.py#L22) to tslearn formats (shouldn't be too hard).

**Comment by fonnesbeck** [comment]
Can scikit-learn's `silhouette_score` not be used directly with `tslearn.metrics.soft_dtw` as the metric?

**Comment by rtavenar** [comment]
Hum, a first problem we would have is that we don't expect the same format for parameter `X` and `sklearn` checks that the provided `X` is valid.

Something that might work would be to fool `sklearn` by providing a reshaped `X` array and a metric function that reshapes arrays before computing their similarity. Something like:

```python
n, sz, d = X.shape
sklearn_X = X.reshape((n, -1))
sklearn_metric = lambda x, y: metric_fun(x.reshape(sz, d), y.reshape(sz, d))
```

**Comment by rtavenar** [comment]
I just pushed an attempt to provide this functionality. Could you guys test it and give some feedback about it?

**Comment by rtavenar** [comment]
@jonpappalord @fonnesbeck 
Guys, could you tell if the implemented feature matches your needs? If so, I would close this Issue.

**Comment by fonnesbeck** [comment]
I'm testing it now, but I'm using a gigantic database, so it will take a few hours to complete.

**Comment by rtavenar** [comment]
Since I have no news, I close this issue, re-open it if needed.

**Comment by ceydaakbulut95** [comment]
Hello. Good to see this comments :) I have tried what you coded and it worked! Thanks!


mySeries=np.array(mySeries)
n, sz, d = mySeries.shape
sklearn_X = mySeries.reshape((n, -1))
sklearn_metric = lambda x, y: metric_fun(x.reshape((sz, d)), y.reshape((sz, d)))
km.fit_predict(sklearn_X)
score = silhouette_score(sklearn_X, km.labels_, metric='euclidean')

print('Silhouetter Score: %.3f' % score)
